benchmark/benchmark.py

```diff
@@ -0,0 +1,143 @@
+import numpy as np
+import pandas as pd
+
+import torch
+from ptflops import get_model_complexity_info
+
+
+memory_unit_constants = {'GB': 1e-9, 'MB': 1e-6, 'KB': 1e-3, 'B': 1.}
+
+
+class Time_tracker:
+    def __enter__(self):
+        self.start = torch.cuda.Event(enable_timing=True)
+        self.end = torch.cuda.Event(enable_timing=True)
+        self.start.record()
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.end.record()
+        torch.cuda.synchronize()
+        
+    def timer(self):
+        return self.start.elapsed_time(self.end)
+
+
+class Memory_tracker:
+    def __init__(self, device=None):
+        self.device = device
+        
+    def __enter__(self):
+        self.start_memory = get_memory(device=self.device)
+        return self
+
+    def __exit__(self, type, value, traceback):
+        self.end_memory = get_memory(reset_memory=False, device=self.device) 
+        
+    def memory_allocated(self):
+        return self.end_memory - self.start_memory 
+
+
+
+def get_memory(reset_memory=True, device=None):
+    """Take current max allocated memory, either with or without resetting"""
+    if reset_memory:
+        torch.cuda.reset_peak_memory_stats()
+            
+    max_memory = torch.cuda.max_memory_allocated(device)
+        
+    return max_memory
+
+
+def make_initialization_inputs(inputs, device=None):
+    """ Take either tensor, shape tuple or list of them, and always return tensor or list of them. """
+    if isinstance(inputs, torch.Tensor):
+        pass
+    elif isinstance(inputs, tuple):
+        inputs = torch.rand(*inputs, device=device)
+    elif isinstance(inputs, list):
+        inputs = [make_initialization_inputs(item, device=device) for item in inputs]
+    return inputs
+
+    
+
+def tracker(module, inputs, repeats=300, warmup=40, device=None, track_backward=True, # .py file ???
+            channels_last=False, amp=False, memory_unit='MB') -> dict:
+    """Track module #macs, #parameters, time and memory consumption on forward and backward pass for a given inputs tensor or inputs shape"""
+    
+    memory_unit_constant = memory_unit_constants[memory_unit]
```

Comment:
Think more about newlines, dev-comments and overall sequence of actions in the function